### PR TITLE
added support for old bugged refs/tags

### PR DESCRIPTION
**Pull request by mjksmith**
Bugged == of the format 'paasta-paasta-cluster.instance'.

Also added an itest that makes sure generate deployments works properly with force_bounce and state tags.

### CODE REVIEW

File: general_itests/deployments_json.feature [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Feature: Per-Service Deployments.json can be written and read back
 
     Scenario: Per-Service Deployments.json can be written and read back
-        Given a test git repo is setup with commits
-        When we generate deployments.json for that service
+       Given a test git repo is setup with commits
+	When paasta stop is run against the repo
```
Comment (solarkennedy) [inline general_itests/deployments_json.feature]: Thanks! This is good stuff.

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 2]
```diff
@@ -127,12 +134,13 @@ def issue_state_change_for_branches(service, instance, cluster, branches, force_
...
     """Requests a change of state to start or stop given branches of a service."""
-    service = figure_out_service_name(args)
     instance = args.instance
     cluster = args.cluster
+    soa_dir = args.soa_dir
+    service = figure_out_service_name(args, soa_dir)
 
     branch = "paasta-%s.%s" % (cluster, instance)
-    all_branches = list(get_branches(service))
+    all_branches = list(get_branches(service, soa_dir))
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/start_stop_restart.py]: can you kwargs this please

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 3]
```diff
@@ -127,12 +134,13 @@ def issue_state_change_for_branches(service, instance, cluster, branches, force_
 
 def paasta_start_or_stop(args, desired_state):
     """Requests a change of state to start or stop given branches of a service."""
-    service = figure_out_service_name(args)
     instance = args.instance
     cluster = args.cluster
+    soa_dir = args.soa_dir
+    service = figure_out_service_name(args, soa_dir)
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/start_stop_restart.py]: also here, kwargs

**Comment by solarkennedy** [comment]
fix + ship. I would like to get @EvanKrall's official opinion on this whole paasta-, paasta-paasta- nonsense.

If the itest paases, then wfm

File: general_itests/steps/deployments_json_steps.py [hunk 4]
```diff
@@ -57,6 +60,30 @@ def step_impl_given(context):
     context.expected_commit = commit.id
 
 
+@when(u'paasta stop is run against the repo')
```
Comment (nhandler) [inline general_itests/steps/deployments_json_steps.py]: It might be nice to parameterize this step so that you can run any paasta command against it.

**Pull request merged**